pkg/controller/drain.go

```diff
@@ -235,7 +235,8 @@ func (o *DrainOptions) daemonsetFilter(pod api.Pod) (bool, *warning, *fatal) {
 	if controllerRef == nil || controllerRef.Kind != "DaemonSet" {
 		return true, nil, nil
 	}
-	if _, err := o.client.Extensions().DaemonSets(pod.Namespace).Get(controllerRef.Name, metav1.GetOptions{}); err != nil {
+	//if _, err := o.client.Extensions().DaemonSets(pod.Namespace).Get(controllerRef.Name, metav1.GetOptions{}); err != nil {
```

Comment:
Please remove commneted code, if its no more required. Ideally i would suggest to move to controller-runtime API, where you pass an object  and call `client.Get()` to retrieve filled object copy.

Reply:
Done. Refer - https://github.com/gardener/machine-controller-manager/pull/398/commits/e2076e9669307b66a27954e466e2f640bfa4da1d.